src/DotLiquid/Util/NumericConverter.cs

```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace DotLiquid.Util
+{
+    /// <summary>
+    /// ExtensionMethods for converting to numeric values.
+    /// </summary>
+    public static class NumericConverter
+    {
+        /// <summary>
+        /// Check if the object is a floating point (real) type.
+        /// </summary>
+        /// <param name="value">The value to check</param>
+        /// <returns>True if the value is decimal, float or double; false otherwise.</returns>
+        public static bool IsReal(object value) => value is double || value is float || value is decimal;
+
+        /// <summary>
+        /// Check if the object is an integer type.
+        /// </summary>
+        /// <param name="value">The value to check</param>
+        /// <returns>True if the value is an integer; false otherwise.</returns>
+        public static bool IsInteger(object value) =>
+            value is int || value is uint || value is long || value is ulong ||
+            value is short || value is ushort || value is byte || value is sbyte;
+
+        /// <summary>
+        /// Check if the object is a numeric type.
+        /// </summary>
+        /// <param name="value">The value to check</param>
+        /// <returns>True if the value is an integer or floating point number; false otherwise.</returns>
+        public static bool IsNumeric(object value) => IsReal(value) || IsInteger(value);
+
+        /// <summary>
+        /// A dictionary of numeric types and their allowable conversions.
+        /// Based on the promotion table at
+        /// https://docs.microsoft.com/en-us/dotnet/standard/base-types/conversion-tables

```

Comment:
```suggestion
        /// https://docs.microsoft.com/dotnet/standard/base-types/conversion-tables

```
By removing the locale URL segment, the user will get whatever locale they configured for their browsing experience.